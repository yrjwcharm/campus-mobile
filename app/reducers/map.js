const initialState = {
	history: [],
	results: null
}

function map(state = initialState, action) {
	const newState = { ...state }

	switch (action.type) {
		case 'UPDATE_HISTORY': {
			newState.history = action.history
			return newState
		}
		case 'SET_SEARCH_RESULTS': {
			newState.results = action.results

			return newState
		}
		default:
			return state
	}
}

module.exports = map
