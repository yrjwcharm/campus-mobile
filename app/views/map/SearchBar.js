import React from 'react'
import { View, TextInput, ActivityIndicator, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import css from '../../styles/css'

class SearchBar extends React.Component {
	constructor(props) {
		super(props)
		this.state = { text: '' }
	}

	render() {
		const {
			reff,
			placeholder,
			update,
			iconStatus,
			style,
			onFocus,
			pressIcon,
			searchInput
		} = this.props

		return (
			<View style={[css.map_searchbar_container, style]}>
				<TouchableOpacity
					style={css.map_searchbar_icon_container}
					onPress={event => pressIcon()}
				>
					<SearchIcon iconStatus={iconStatus} />
				</TouchableOpacity>
				<TextInput
					ref={reff}
					placeholder={placeholder}
					autoCorrect={false}
					onSubmitEditing={event => update(event.nativeEvent.text.trim())}
					onChangeText={text => this.setState({ text })}
					blurOnSubmit={true}
					returnKeyType="search"
					clearButtonMode="while-editing"
					selectTextOnFocus={true}
					style={css.map_searchbar_input}
					underlineColorAndroid="rgba(0,0,0,0)"
					onFocus={event => onFocus()}
					onBlur={() => this.setState({ text: searchInput })}
					defaultValue={searchInput}
					value={this.state.text}
					maxLength={35}
				/>
			</View>
		)
	}
}

const SearchIcon = ({ iconStatus }) => {
	switch (iconStatus) {
		case 'load': 	return (<ActivityIndicator size="small" />)
		case 'search':	return (<Icon name="search" size={24} color="rgba(0,0,0,.5)" />)
		case 'menu':	return (<Icon name="bars" size={24} color="rgba(0,0,0,.5)" />)
		case 'back':	return (<Icon name="arrow-left" size={24} color="rgba(0,0,0,.5)" />)
	}
}

SearchBar.defaultProps = {
	placeholder: 'Search here',
	searchInput: ''
}

export default SearchBar
