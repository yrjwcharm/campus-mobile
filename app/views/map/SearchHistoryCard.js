import React from 'react'
import { View, Text, FlatList } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import css from '../../styles/css'
import Touchable from '../common/Touchable'

const SearchHistoryCard = ({ data, pressHistory, removeHistory }) => (
	<View style={css.shc_list_container}>
		<FlatList
			showsVerticalScrollIndicator={false}
			data={data}
			keyboardShouldPersistTaps="always"
			keyExtractor={(listItem, index) => (String(listItem) + String(index))}
			renderItem={
				({ item: rowData, index: rowID }) => (
					<SearchHistoryItem
						data={rowData}
						index={rowID}
						pressHistory={pressHistory}
						removeHistory={removeHistory}
					/>
				)
			}
		/>
	</View>
)

const SearchHistoryItem = ({ data, index, pressHistory, removeHistory }) => (
	<Touchable onPress={() => { pressHistory(data) }}>
		<View style={css.shc_list_row}>
			<Icon name="history" size={22} style={css.shc_row_icon} />
			<Text style={css.shc_row_text}>{data}</Text>
			<Touchable
				onPress={() => { removeHistory(index) }}
				style={css.shc_row_remove}
			>
				<Icon name="cancel" size={18} style={css.shc_row_icon} />
			</Touchable>
		</View>
	</Touchable>
)

export default SearchHistoryCard
