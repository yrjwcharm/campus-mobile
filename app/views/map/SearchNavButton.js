import React from 'react'
import { TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import css from '../../styles/css'

const SearchNavButton = ({ visible, onPress }) => {
	if (visible) {
		return (
			<TouchableOpacity
				style={[css.snb_container, css.snb_container_nav]}
				onPress={() => onPress()}
			>
				<Icon name="location-arrow" size={20} color="white" />
			</TouchableOpacity>
		)
	} else {
		return null
	}
}

export default SearchNavButton
