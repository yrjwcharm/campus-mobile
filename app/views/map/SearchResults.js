import React from 'react'
import { View,	Text, FlatList } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import css from '../../styles/css'
import Touchable from '../common/Touchable'

const SearchResultsCard = ({ results, onSelect }) => {
	if (results) {
		return (
			<FlatList
				style={css.ms_container}
				data={results}
				keyExtractor={(listItem, index) => (listItem.title + index)}
				renderItem={
					({ item: rowData, index: rowID }) => (
						<SearchResultsItem
							data={rowData}
							index={rowID}
							onSelect={onSelect}
						/>
					)
				}
			/>
		)
	} else {
		return null
	}
}

const SearchResultsItem = ({ data, onSelect, index }) => (
	<Touchable onPress={() => onSelect(index)}>
		<View style={css.ms_result_row}>
			<Icon name="map-marker" size={22} style={css.ms_result_icon} />
			<Text style={css.ms_result_title}>{data.title}</Text>
			{data.distanceMilesStr ? (
				<Text style={css.ms_result_dist}>{data.distanceMilesStr}</Text>
			) : null }
		</View>
	</Touchable>
)

export default SearchResultsCard
