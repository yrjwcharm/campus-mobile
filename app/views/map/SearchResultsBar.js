import React from 'react'
import { View, TouchableOpacity, Text } from 'react-native'
import css from '../../styles/css'

const SearchResultsBar = ({ visible, onPress }) => {
	if (visible) {
		return (
			<View style={css.srb_bottomBarContainer}>
				<TouchableOpacity
					style={css.srb_bottomBarContent}
					onPress={() => onPress()}
				>
					<Text style={css.srb_bottomBarText}>
						See More Results
					</Text>
				</TouchableOpacity>
			</View>
		)
	} else {
		return (null)
	}
}

export default SearchResultsBar
