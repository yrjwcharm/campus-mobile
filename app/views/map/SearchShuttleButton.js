import React from 'react'
import { TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import css from '../../styles/css'
import COLOR from '../../styles/ColorConstants'

const SearchShuttleIcon = ({ visible, onPress }) => {
	if (visible) {
		return (
			<TouchableOpacity
				onPress={() => onPress()}
				style={css.snb_container}
			>
				<Icon name="bus" size={20} color={COLOR.WHITE} />
			</TouchableOpacity>
		)
	} else {
		return null
	}
}

export default SearchShuttleIcon
