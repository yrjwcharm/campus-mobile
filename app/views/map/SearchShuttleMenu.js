import React from 'react'
import {
	View,
	Text,
	StyleSheet,
	FlatList,
	TouchableOpacity
} from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'
import COLOR from '../../styles/ColorConstants'
import LAYOUT from '../../styles/LayoutConstants'

const SearchShuttleMenu = ({ onToggle, toggles, shuttle_routes }) => (
	<View style={styles.card_main}>
		<View style={styles.list_container}>
			{shuttle_routes ? (
				<MenuList
					shuttles={shuttle_routes}
					onToggle={onToggle}
					toggles={toggles}
				/>
			) : null }
		</View>
	</View>
)

const MenuList = ({ shuttles, onToggle, toggles }) => (
	<FlatList
		data={Object.values(shuttles)}
		keyExtractor={(listItem, index) => (String(listItem.id) + String(index))}
		renderItem={
			({ item: rowData, index: rowID }) => (
				<MenuItem
					data={rowData}
					index={rowID}
					onToggle={onToggle}
					state={toggles[rowData.id]}
				/>
			)
		}
	/>
)

const MenuItem = ({ data, index, onToggle, state }) => (
	<TouchableOpacity
		onPress={() => onToggle(data.id)}
		value={state}
		style={styles.list_row}
	>
		<Text style={{ flex: 4 }}>
			{data.name.trim()}
		</Text>
		<Icon
			style={styles.radio_icon}
			name={state ? 'ios-radio-button-on' : 'ios-radio-button-off'}
			size={20}
			color={COLOR.SECONDARY}
		/>
	</TouchableOpacity>
)

// Port to CSS/cleanup
// device - (statusBar + navHeight + searchBar + listPadding + tabBar)
const listHeight = LAYOUT.WINDOW_HEIGHT - (LAYOUT.STATUS_BAR_HEIGHT + LAYOUT.NAVIGATOR_HEIGHT + LAYOUT.TAB_BAR_HEIGHT + 44 + 16 + 20) // 18 + 64 + (44 * getPRM()));

const styles = StyleSheet.create({
	list_container: { width: LAYOUT.MAX_CARD_WIDTH, maxHeight: listHeight },
	card_main: { top: 50, backgroundColor: 'white', margin: 6, alignItems: 'flex-start', justifyContent: 'center', borderRadius: 3 },
	list_row: { alignItems: 'center', justifyContent: 'center', flexDirection: 'row', paddingVertical: 14, paddingHorizontal: 8, borderBottomWidth: 1, borderBottomColor: COLOR.MGREY, overflow: 'hidden',  },
	switch_container: { flex: 1, alignItems: 'flex-end' },
	radio_icon: { alignSelf: 'flex-end', marginLeft: 10 },
})

export default SearchShuttleMenu
