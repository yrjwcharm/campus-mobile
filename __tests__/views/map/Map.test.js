import 'react-native'
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// Component to be tested
import { Map } from '../../../app/views/map/Map'

Enzyme.configure({ adapter: new Adapter() })

// Mock props passed down from state
const initialState = {
	location: {
		coords: {
			latitude: 32.88,
			longitude: -117.234
		}
	}
}

// Set up component to be rendered
function setup(props) {
	return shallow(<Map {...props} />)
}

test('renders without crashing', () => {
	const tree = setup(initialState)
	expect(tree).toMatchSnapshot()
})
