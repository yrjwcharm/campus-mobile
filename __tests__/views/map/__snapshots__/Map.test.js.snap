// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View>
  <SearchNavButton
    onPress={[Function]}
    visible={false}
  />
  <SearchShuttleIcon
    onPress={[Function]}
    visible={true}
  />
  <SearchBar
    iconStatus="search"
    onFocus={[Function]}
    placeholder="Search here"
    pressIcon={[Function]}
    reff={[Function]}
    searchInput={null}
    update={[Function]}
  />
  <ScrollView
    keyboardShouldPersistTaps="always"
    scrollEnabled={false}
    showsVerticalScrollIndicator={false}
  >
    <View
      style={
        Object {
          "height": 1250,
        }
      }
    >
      <SearchMap
        location={
          Object {
            "coords": Object {
              "latitude": 32.88,
              "longitude": -117.234,
            },
          }
        }
        selectedResult={null}
        vehicles={Object {}}
      />
    </View>
    <View
      style={
        Object {
          "height": 1250,
        }
      }
    >
      <SearchResultsCard
        onSelect={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "height": 1250,
        }
      }
    >
      <SearchSuggest
        onPress={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "height": 1250,
        }
      }
    >
      <SearchShuttleMenu
        onToggle={[Function]}
      />
    </View>
  </ScrollView>
  <SearchResultsBar
    onPress={[Function]}
    visible={false}
  />
</View>
`;
